prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -237,25 +237,42 @@ class ReportFunction {
             "${service.organization.description} (${service.name})"
         else
             service.organization.description
-        val event = when {
+        val loggerMsg: String
+        when {
             validatedRequest.options == Options.SkipSend -> {
-                ReportEvent(Event.EventAction.NONE, report.id)
+                val event = ReportEvent(Event.EventAction.NONE, report.id)
+                workflowEngine.dispatchReport(event, report, txn)
+                actionHistory.trackCreatedReport(event, report, service)
```

Comment:
I am working toward combining `dispatchReport` and `trackCreatedReport` in an upcoming future PR, so this repeated pattern of the two called together, will get cleaner.

Reply:
Nice.